--- a/backport-include/linux/compiler_attributes.h
+++ b/backport-include/linux/compiler_attributes.h
@@ -39,7 +39,10 @@
  *   gcc: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=108896
  * clang: https://reviews.llvm.org/D148381
  */
-#if __has_attribute(__element_count__)
+/* If the kernel config will provide __counted_by, don't define it here. */
+#ifdef CONFIG_CC_HAS_COUNTED_BY
+/* kernel provides __counted_by */
+#elif __has_attribute(__element_count__)
 # define __counted_by(member)		__attribute__((__element_count__(#member)))
 #else
 # define __counted_by(member)
