--- a/arch/arm/include/asm/system.h
+++ b/arch/arm/include/asm/system.h
@@ -658,6 +658,7 @@ void mmu_set_region_dcache_behaviour(phy
  * Return: 0 if OK
  */
 int noncached_init(void);
+void noncached_set_region(void);

 phys_addr_t noncached_alloc(size_t size, size_t align);
 #endif /* CONFIG_SYS_NONCACHED_MEMORY */
--- a/cmd/cache.c
+++ b/cmd/cache.c
@@ -10,6 +10,7 @@
 #include <command.h>
 #include <cpu_func.h>
 #include <linux/compiler.h>
+#include <asm/system.h>
